sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/src/Listeners/ServiceBusTargetScaler.cs

```diff
@@ -0,0 +1,83 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using Azure.Messaging.ServiceBus;
+using Azure.Messaging.ServiceBus.Administration;
+using Microsoft.Azure.WebJobs.Extensions.ServiceBus.Config;
+using Microsoft.Azure.WebJobs.Host.Scale;
+using Microsoft.Azure.WebJobs.ServiceBus;
+using Microsoft.Azure.WebJobs.ServiceBus.Listeners;
+using Microsoft.Extensions.Logging;
+using System;
+using System.Globalization;
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.WebJobs.Extensions.ServiceBus.Listeners
+{
+    public class ServiceBusTargetScaler : ITargetScaler
+    {
+        private readonly string _functionId;
+        private readonly ServiceBusMetricsProvider _serviceBusMetricsProvider;
+        private readonly ServiceBusOptions _options;
+        private readonly bool _isSessionsEnabled;
+        private readonly TargetScalerDescriptor _targetScalerDescriptor;
+        private readonly string _entityPath;
+        private readonly ILogger _logger;
+
+        public ServiceBusTargetScaler(
+            string functionId,
+            string entityPath,
+            ServiceBusEntityType entityType,
+            Lazy<ServiceBusReceiver> receiver,
+            Lazy<ServiceBusAdministrationClient> administrationClient,
+            ServiceBusOptions options,
+            bool isSessionsEnabled,
+            ILoggerFactory loggerFactory
+            )
+        {
+            _functionId = functionId;
+            _serviceBusMetricsProvider = new ServiceBusMetricsProvider(entityPath, entityType, receiver, administrationClient, loggerFactory);
```

Comment:
@alrod  do you think that will be better to have the ServiceBusMetricsProvider as interface? with that we can have an abstraction level in this constructor and will be easy to create unit tests.

Reply:
We're trying to minimize public surface area and don't want a new public type for this abstraction, that's why it's internal.